testing.md

```diff
@@ -0,0 +1,38 @@
+# Front-End Testing
+
+Extol the [testing pyramid](https://docs.google.com/presentation/d/1NFjgq5SbQ4crNi06SplyqZQ-K9o5RX966AoRVEB38Yg/pub?start=false&loop=false&delayms=3000&slide=id.g18ef6d797_047).
+
+[![Testing pyramid](http://i.imgur.com/1hQdxod.png)](https://docs.google.com/presentation/d/1NFjgq5SbQ4crNi06SplyqZQ-K9o5RX966AoRVEB38Yg/pub?start=false&loop=false&delayms=3000&slide=id.g18ef6d797_047)
+
+## Unit tests
+
+If your code has logic, it should have unit tests. As seen in the pyramid, unit tests should comprise the majority of your tests. Why? Unit tests can be executed quickly, they can be easily automated, they're maintainable and they promote good practices by requiring your code to be written in concise, modular chunks.
+
+A unit test verifies that a single discrete piece of your codebase works as expected. That "piece" could be a JavaScript function, a CommonJS/AMD module or some other unit of functionality. It's vaguely defined but don't let that scare you. Start by choosing one of the test frameworks below and writing tests for individual JavaScript methods. There are many guides about [writing](https://github.com/nelsonic/learn-mocha) [JavaScript](http://alistapart.com/article/writing-testable-javascript) [unit](http://www.smashingmagazine.com/2012/06/27/introduction-to-javascript-unit-testing/) [tests](http://www.htmlgoodies.com/beyond/javascript/testing-javascript-using-the-jasmine-framework.html) that serve as great introductions. At CFPB we like to separate our JavaScript into discrete [npm modules](https://github.com/cfpb/front-end/blob/master/npm.md). This makes testing easier because each module includes self-contained tests.
+
+### Test frameworks
+- [Mocha](http://mochajs.org) is a slick, simple testing framework for Node and browser JavaScript with asynchronous support and a [slew of reporters](http://mochajs.org/#reporters), including [TAP](http://en.wikipedia.org/wiki/Test_Anything_Protocol) output. See the [node-mocha example](examples/node-mocha) for usage.
```

Comment:
Good catch, that line was taken from the 18f guide linked to at the bottom of the list.